instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt

```diff
@@ -16,4 +17,29 @@ class SanityUiRouteTest : SimpleMapViewNavigationTest() {
         Espresso.onIdle()
         arrivalIdlingResource.unregister()
     }
+
+    @Test
+    fun navigation_camera_mode_changes_completes() {
```

Comment:
I reverted the changes introduced in this PR and the provided test does not reproduce a StackOverflow on a Pixel 2 running Android 11 for me. Looking more into it.

Reply:
Is this test crashing on your device without the other changes from the PR? What device and Android version are you running on?

Reply:
It seems this recursion has also been fixed on Android side here: https://cs.android.com/android/_/android/platform/frameworks/base/+/3dbaae1ef4f221b3626810f4ba19eec068dd6304
I'm not sure in what Android it was fixed, 6.0.1 has this bug.

Reply:
Looks like this is it, I was able to reproduce on Android 21 :+1: I'll tweak the PR to make sure that the test targets a specific API level and that we have a device on the farm that can catch this.